### PR TITLE
Emit warnings on conflicts between openrtb 2.6 pre GPP privacy and GPP

### PR DESCRIPTION
**Pull request by hhhjort**

### CODE REVIEW

File: privacy/ccpa/policy.go [hunk 1]
```diff
@@ -29,13 +36,20 @@ func ReadFromRequestWrapper(req *openrtb_ext.RequestWrapper, gpp gpplib.GppConta
 				}
 			}
 		}
+		if req.BidRequest.Regs.USPrivacy != "" {
+			if consent == "" {
+				consent = req.BidRequest.Regs.USPrivacy
```
Comment (AlexBVolcy) [inline privacy/ccpa/policy.go:41]: Can you add test coverage for this line?

File: endpoints/openrtb2/auction.go [hunk 2]
```diff
@@ -1609,29 +1623,41 @@ func (deps *endpointDeps) validateApp(req *openrtb_ext.RequestWrapper) error {
...
-		}
+	if req.User.Geo != nil && req.User.Geo.Accuracy < 0 {
+		return append(errL, errors.New("request.user.geo.accuracy must be a positive number"))
 	}
 
+	if req.User.Consent != "" {
+		for _, sec := range gpp.Sections {
+			if sec.GetID() == constants.SectionTCFEU2 && sec.GetValue() != req.User.Consent {
+				errL = append(errL, &errortypes.Warning{
+					Message:     "user.consent GDPR string conflicts with GPP (regs.gpp) GDPR string and will be ignored",
```
[the remark this answers is not in the document]
Comment (hhhjort) [reply endpoints/openrtb2/auction.go:1642]: Fixed the wording, I haven't checked Java yet.

File: endpoints/openrtb2/auction.go [hunk 3]
```diff
@@ -1609,29 +1623,41 @@ func (deps *endpointDeps) validateApp(req *openrtb_ext.RequestWrapper) error {
...
-	if req != nil && req.BidRequest != nil && req.User != nil {
-		if req.User.Geo != nil && req.User.Geo.Accuracy < 0 {
-			return errors.New("request.user.geo.accuracy must be a positive number")
-		}
+	if req.User.Geo != nil && req.User.Geo.Accuracy < 0 {
+		return append(errL, errors.New("request.user.geo.accuracy must be a positive number"))
 	}
 
+	if req.User.Consent != "" {
+		for _, sec := range gpp.Sections {
```
[the remark this answers is not in the document]
Comment (hhhjort) [reply endpoints/openrtb2/auction.go:1639]: I am fine with that.

File: endpoints/openrtb2/auction_test.go [hunk 4]
```diff
@@ -3529,15 +3529,40 @@ func TestIOS14EndToEnd(t *testing.T) {
...
+			file:            "us-privacy-conflict.json",
+			expectedWarning: "regs.us_privacy consent does not match uspv1 in GPP and will be ignored",
+		},
+		{
+			name:            "gdpr-conflict", // gdpr signals do not match
+			file:            "gdpr-conflict.json",
+			expectedWarning: "regs.gdpr signal conflicts with GPP (regs.gpp_sid) and will be ignored",
+		},
+		{
+			name:            "gdpr-conflict2", // gdpr consent strings do not match
```
[the remark this answers is not in the document]
Comment (hhhjort) [reply endpoints/openrtb2/auction_test.go:3553]: Naming is an art. :)

File: privacy/ccpa/policy.go [hunk 5]
```diff
@@ -29,13 +36,20 @@ func ReadFromRequestWrapper(req *openrtb_ext.RequestWrapper, gpp gpplib.GppConta
...
+			} else {
+				if consent != req.BidRequest.Regs.USPrivacy {
+					warn = &errortypes.Warning{
+						Message:     "regs.us_privacy consent does not match uspv1 in GPP and will be ignored",
+						WarningCode: errortypes.InvalidPrivacyConsentWarningCode}
+				}
```
[the remark this answers is not in the document]
Comment (hhhjort) [reply privacy/ccpa/policy.go:47]: Yes

File: privacy/ccpa/policy.go (hunk 1 above)
Comment (hhhjort) [reply privacy/ccpa/policy.go:41]: Test added

File: endpoints/openrtb2/auction_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (SyntaxNode) [reply endpoints/openrtb2/auction_test.go]: https://www.reddit.com/r/ProgrammerHumor/comments/9ccjvk/use_uuids_instead_of_variable_names/

**Review by AlexBVolcy — APPROVED** [review]
LGTM

**Pull request merged**